apps/src/dance/lab2/ProgramExecutor.ts

```diff
@@ -117,14 +116,19 @@ export default class ProgramExecutor {
 
     this.hooks.runUserSetup();
 
-    // Force preview draw to occur **after** any
-    // draw iterations already queued up.
-    // redraw() (rather than draw()) is p5's recommended way
-    // of drawing once.
-    setTimeout(() => {
+    const previewDraw = () => {
```

Comment:
I'll likely end up changing this since we'll be using live preview here, but for consistency should we move `runUserSetup` into the preview callback here as well?

Reply:
Yep will do!

Reply:
@sanchitmalhotra126 a little refactor of lab2 code, lmk what you think. Not sure how to test the main `execute()` path manually, but I checked that the preview in the modal still renders.

https://github.com/code-dot-org/code-dot-org/pull/54456/commits/c9a80832c1faa1264cef3d68159be79989d0c35b

Reply:
Sure thing, for my understanding what was the reasoning behind splitting up `preloadSpritesAndCompileCode`? Do we need to perform these independently, or just to separate concerns?

Reply:
It didn't like having an async function (loading of sprites) in the draw callback, which I think makes sense?

Reply:
Gotcha, doesn't look like we're calling preloadSprites or compileCode within the draw callback through right?

Reply:
🤦 you're totally right, had something in my mind from working on the legacy dance implementation that wasn't true, distracted brain not fully functioning. Reverted back to the simpler approach!

Reply:
No worries, thanks for the clarification!